src/pymor/reductors/data_driven.py

```diff
@@ -0,0 +1,268 @@
+    problems and returns a suitable model for the respective case.
+
+    Depending on the argument `target_quantity`, this reductor
+    either approximates the solution or the output as a parametric
+    quantity by training a machine learning surrogate.
+
+    In case of an approximation of the solution, the reductor trains
+    a machine learning regressor that approximates the mapping from
+    parameter space to a |NumPy array|. Typically, the array contains
+    coefficients of the solution with respect to a reduced basis.
+
+    For `target_quantity='output'`, the machine learning regressor
+    directly approximates the output depending on the parameter.
+    The outputs for the training parameters are either computed using
+    the full-order model or can be provided as the `training_snapshots`
+    argument.
+
+    Parameters
+    ----------
+    training_parameters
+        |Parameter values| to use for training of the regressor.
+    training_snapshots
+        Iterable containing the training snapshots of the regressor.
+        Contains the solutions (reduced coefficients w.r.t. the reduced basis
+        or outputs associated to the parameters in `training_parameters`.
+        In the case of a time-dependent problem, the snapshots are assumed to be
+        equidistant in time.
+    regressor
+        Regressor with `fit` and `predict` methods similar to scikit-learn
+        regressors that is trained in the `reduce`-method.
+    target_quantity
+        Either `'solution'` or `'output'`, determines which quantity to learn.
+    T
+        In the instationary case, determines the final time until which to solve.
+    time_vectorized
+        In the instationary case, determines whether to predict the whole time
+        trajectory at once (time-vectorized version; output of the regressor is
+        typically very high-dimensional in this case) or if the result for a
+        single point in time is approximated (time serves as an additional input
+        to the regressor).
+    output_functional
+        |Operator| mapping a given solution to the model output. In many applications,
+        this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+        This is not required, however.
+    input_scaler
+        If not `None`, a scaler object with `fit`, `transform` and
+        `inverse_transform` methods similar to the scikit-learn interface can be
+        used to scale the parameters before passing them to the regressor.
+    output_scaler
+        If not `None`, a scaler object with `fit`, `transform` and
+        `inverse_transform` methods similar to the scikit-learn interface can be
+        used to scale the outputs (reduced coeffcients or output quantities)
+        before passing them to the regressor.
+    """
+
+    def __init__(self, training_parameters, training_snapshots,
+                 regressor=VKOGARegressor(GaussianKernel()), target_quantity='solution',
```

Comment:
`GaussianKernel()` not required as `VKOGARegressor` uses this as default.

Reply:
Further `VKOGARegressor` is a mutable default argument, which is evaluated once at function definition time, but not each time the function is called. I think this becomes problematic in case multiple `DataDrivenReductor`s get initialized without explicitly specifying the regressor argument, as they than share the same regressor object and calling reduce (which calls fit on the regressor) on one reductor affects the regressor in the other reductor. I think its better to say `reductor=None` as init argument and then initialize the regessor in the `__init__` if `regressor=None`.